phonebook-ui/src/app/contact-details/phone/phone-element/phone-element.component.ts

```diff
@@ -0,0 +1,82 @@
+import {Component, Input, OnDestroy, OnInit} from '@angular/core';
+import {Phone} from "../../../model/phone";
+import {Subscription} from "rxjs";
+import {PhoneService} from "../../../service/phone.service";
+import {ActivatedRoute, Router} from "@angular/router";
+import {HttpErrorResponse} from "@angular/common/http";
+import {FormControl, FormGroup, Validators} from "@angular/forms";
+
+@Component({
+  selector: 'app-phone-element',
+  templateUrl: './phone-element.component.html',
+  styleUrls: ['./phone-element.component.css', '../../contact-details.component.css']
+})
+export class PhoneElementComponent implements OnInit, OnDestroy {
+  private subscriptions: Subscription[] = [];
+  getAllPhoneErrorMessage: string | undefined;
+  phoneEditFlag = false;
+  contactId: number | undefined;
+  flag: boolean | undefined;
+  updatePhoneForm!: FormGroup;
+  logError: String | undefined;
+  @Input()
+  phone: Phone | undefined;
+
+
+  constructor(private phoneService: PhoneService, private route: ActivatedRoute, private router: Router) {
+  }
+
+
+  ngOnInit(): void {
+    this.contactId = Number(this.route.snapshot.paramMap.get('id'));
+    this.flag = this.phone?.isFavorite;
+    this._initFrom();
+  }
+
+  private _initFrom() {
+    this.updatePhoneForm = new FormGroup({
+      countryCode: new FormControl(this.phone?.countryCode, Validators.required),
+      telephoneNumber: new FormControl(this.phone?.telephoneNumber, Validators.required),
+      favorite: new FormControl(this.flag),
```

Comment:
favorite: new FormControl(**this.flag**) ->  favorite: new FormControl(**this.phone?.isFavorite**)
???